pkg/apis/core/types.go

```diff
@@ -544,6 +566,41 @@ const (
 	PersistentVolumeClaimNodeResizeFailed ClaimResourceStatus = "NodeResizeFailed"
 )
 
+// +enum
+type PersistentVolumeClaimModifyVolumeStatus string
+
+const (
+	// Pending indicates that the PersistentVolumeClaim cannot be modified due to requirements not being met, such as
+	// the specified VolumeAttributesClass doesn't exist
+	PersistentVolumeClaimControllerModifyVolumePending PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumePending"
+	// State set when modify volume controller starts modifying the volume
+	PersistentVolumeClaimControllerModifyVolumeInProgress PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumeInProgress"
+	// State set when modify volume has failed in modify volume controller with a terminal error.
+	// When the request has been rejected as invalid by the CSI driver. To resolve this error,
+	// the PersistentVolumeClaim needs to specify a valid VolumeAttributesClass.
+	PersistentVolumeClaimControllerModifyVolumeInfeasible PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumeInfeasible"
+)
+
+// ModifyVolumeStatus represents the status object of ControllerModifyVolume operation
+type ModifyVolumeStatus struct {
+	// CurrentVolumeAttributesClassName is the current name of the VolumeAttributesClass the PVC is using
```

Comment:
When we refer to field names in the API docs, it should follow json camelCase syntax.